fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/event/AbstractEventCreationFragment.java

```diff
@@ -27,34 +28,37 @@
  * time.
  *
  * <p>This class handles these fields.
- *
- * <p>TODO: this class needs to be refactored
  */
 public abstract class AbstractEventCreationFragment extends Fragment {
 
-  public static final int START_DATE_REQUEST_CODE = 11; // Used to identify the request
-  public static final int END_DATE_REQUEST_CODE = 12;
-  public static final int START_TIME_REQUEST_CODE = 13;
-  public static final int END_TIME_REQUEST_CODE = 14;
-  public static final String NO_LOCATION = "";
+  public static final String START_DATE_REQUEST_KEY = "START_DATE"; // Used to identify the request
+  public static final String END_DATE_REQUEST_KEY = "END_DATE";
+  public static final String START_TIME_REQUEST_KEY = "START_TIME";
+  public static final String END_TIME_REQUEST_KEY = "END_TIME";
+
+  public static final long CREATION_TIME_IN_SECONDS = Instant.now().getEpochSecond();
+
   public final DateFormat DATE_FORMAT = new SimpleDateFormat("dd/MM/yyyy", Locale.FRENCH);
   public final DateFormat TIME_FORMAT = new SimpleDateFormat("HH:mm", Locale.FRENCH);
-  public static final long CREATION_TIME_IN_SECONDS = Instant.now().getEpochSecond();
-  public static long startTimeInSeconds;
-  public static long endTimeInSeconds;
-  private Calendar startDate;
-  private Calendar endDate;
-  private Calendar startTime;
-  private Calendar endTime;
+
   private final Calendar today = Calendar.getInstance();
   private final Calendar completeStartTime = Calendar.getInstance();
   private final Calendar completeEndTime = Calendar.getInstance();
+
+  public long startTimeInSeconds;
+  public long endTimeInSeconds;
```

Comment:
Sonarcloud suggested that we should provide accessors for theses.
Maybe just using protected instead of public could also work.

Reply:
Done